Dd 13 refactor jira response models (without the JiraIssue file)

georgiantanase opened the pull request:
Refactored all the jira response models except the ones in JiraIssue file. It's a big one and I decided to do it in another branch

Commit:
DD-13
- refactor jira response models to be serializable and to have the properties with the first letter in upperCase

Commit:
DD-13
- update csproj file and fix build problems

Sebyd commented on Equilobe.DailyReport.Models/Jira/Entries.cs:
```diff
@@ -1,40 +1,72 @@
+                return Convert.ToDateTime(Created);
+            }
+            set
+            {
```
remove the setters if they are empty

georgiantanase replied:
The properties are used throughout the project, and if I remove the set, even though it's empty, it would mean that the property is read only. Having an empty set, means that I can set the property's value outside the class.

Sebyd replied:
Is this set outside this? For me it seems that we only need the getter.

georgiantanase replied:
Yes, it is used in IssueAdapter

Sebyd commented on JiraReporter/IssueAdapter.cs:
```diff
@@ -22,7 +22,7 @@ public static void RemoveWrongEntries(EntryContext context)
                 return;
 
-            context.Issue.Entries.RemoveAll(e => e.StartDate.ToOriginalTimeZone(context.OffsetFromUtc) < context.FromDate || e.StartDate.ToOriginalTimeZone(context.OffsetFromUtc) >= context.ToDate || context.AuthorName != e.AuthorFullName);
+            context.Issue.Entries.RemoveAll(e => e.StartedAt.ToOriginalTimeZone(context.OffsetFromUtc) < context.FromDate || e.StartedAt.ToOriginalTimeZone(context.OffsetFromUtc) >= context.ToDate || context.AuthorName != e.AuthorFullName);
```
let's refactor this for better readability.
something like:
context.Issue.Entries
.RemoveAll(e => e.StartedAt.ToOriginalTimeZone(context.OffsetFromUtc) < context.FromDate)
.RemoveAll(e => e.StartedAt.ToOriginalTimeZone(context.OffsetFromUtc) >= context.ToDate)
.RemoveAll(e=> e.AuthorFullName != context.AuthorName);

georgiantanase replied:
RemoveAll is a linq method which returns an int, so I can't have multiple removeAll, linked together. I can put each condition on a new line for readability, if that's ok.

georgiantanase replied:
Or I can create an extension method, which will allow to make multiple Remove calls.

Sebyd replied:
Yes, create an extension method then.

Sebyd commented on JiraReporter/Services/AuthorLoader.cs:
```diff
@@ -282,7 +282,7 @@ private void AdjustSubtasks(IssueDetailed issue)
             foreach (var subtask in issue.SubtasksDetailed)
             {
-                subtask.Entries.RemoveAll(e => e.AuthorFullName != _currentAuthor.Name || e.StartDate < _context.FromDate || e.StartDate > _context.ToDate);
+                subtask.Entries.RemoveAll(e => e.AuthorFullName != _currentAuthor.Name || e.StartedAt < _context.FromDate || e.StartedAt > _context.ToDate);
```
same as I have suggested before.

georgiantanase replied:
Same as I said before.

Sebyd commented on JiraReporter/Services/DayLogLoader.cs:
```diff
@@ -71,7 +71,7 @@ private void AddIssueToDaylog(JiraDayLog dayLog, IssueDetailed issue)
             currentIssue.ExistsInTimesheet = true;
 
-            currentIssue.Entries = issue.Entries.Where(e => e.AuthorFullName == _author.Name && e.StartDate.ToOriginalTimeZone(_context.OffsetFromUtc).Date >= _date && e.StartDate.ToOriginalTimeZone(_context.OffsetFromUtc) < _date.AddDays(1)).ToList(); 
+            currentIssue.Entries = issue.Entries.Where(e => e.AuthorFullName == _author.Name && e.StartedAt.ToOriginalTimeZone(_context.OffsetFromUtc).Date >= _date && e.StartedAt.ToOriginalTimeZone(_context.OffsetFromUtc) < _date.AddDays(1)).ToList(); 
```
let's also separate this where into multiple queries based on && (same as we do for the RemoveAll stuff). foreach && we add a new .Where query instead.

Sebyd replied:
the Where should work as suggested.

Sebyd commented on JiraReporter/Services/DayLogLoader.cs:
```diff
@@ -87,7 +87,7 @@ private void SetSubtaskEntries(IssueDetailed issue)
             foreach(var subtask in issue.SubtasksDetailed)
             {
-                subtask.Entries = subtask.Entries.Where(e => e.AuthorFullName == _author.Name && e.StartDate.ToOriginalTimeZone(_context.OffsetFromUtc).Date >= _date && e.StartDate.ToOriginalTimeZone(_context.OffsetFromUtc) < _date.AddDays(1)).ToList();
+                subtask.Entries = subtask.Entries.Where(e => e.AuthorFullName == _author.Name && e.StartedAt.ToOriginalTimeZone(_context.OffsetFromUtc).Date >= _date && e.StartedAt.ToOriginalTimeZone(_context.OffsetFromUtc) < _date.AddDays(1)).ToList();
```
same as suggested in the upper comment

Commit:
DD-13
- break linq methods into multiple calls

Commit:
DD-13
- create a linq extension method to remove elements in a list based on a predicate and return the new list, to allow multiple removals based on multiple criteria
